pkg/subctl/cmd/join.go

```diff
@@ -95,6 +97,11 @@ func addJoinFlags(cmd *cobra.Command) {
 	cmd.Flags().BoolVar(&natTraversal, "natt", true, "enable NAT traversal for IPsec")
 	cmd.Flags().BoolVar(&disableNat, "disable-nat", false, "disable NAT for IPsec")
 	err := cmd.Flags().MarkDeprecated("disable-nat", "please use --natt=false instead")
+
+	cmd.Flags().BoolVar(&preferredServer, "preferred-server", false,
+		"enable this cluster as a preferred server for dataplane connections")
+	cmd.Flags().BoolVar(&forceUDPEncaps, "force-udp-encaps", false, "force UDP encapsulation for IPSec")
```

Comment:
If NatDiscovery figures out that there is no NAT between the clusters (so ideally they can communicate using ESP) and user passed `--force-udp-encaps`, does it cause IPsec to use UDP Encapsulation?

Reply:
yes, it forces to always use encapsulation, even if there is no nat

That's the purpose :-)

Reply:
Okay, thanks for the clarification.